test/jdk/sun/security/pkcs11/PKCS11Test.java

```diff
@@ -828,6 +832,58 @@ static List<String> getSupportedAlgorithms(String type, String alg,
         return algorithms;
     }
 
+    private static final SecureRandom srdm = new SecureRandom();
+
+    static SecretKey generateKey(String alg, int keySize) {
+        if (alg.contains("PBE")) {
+            return generateKeyPBE(alg, keySize);
+        } else {
+            return generateKeyNonPBE(alg, keySize);
+        }
+    }
+
+    private static SecretKey generateKeyNonPBE(String alg, int keySize) {
+        byte[] keyVal = new byte[keySize];
+        srdm.nextBytes(keyVal);
+        return new SecretKeySpec(keyVal, alg);
+    }
+
+    private static SecretKey generateKeyPBE(String alg, int keySize) {
+        char[] pass = new char[keySize];
+        for (int i = 0; i < pass.length; i++) {
+            pass[i] = (char) ('0' + srdm.nextInt(74));
+        }
+        byte[] salt = new byte[srdm.nextInt(8, 16)];
+        srdm.nextBytes(salt);
+        int iterations = srdm.nextInt(1, 1000);
+        return new javax.crypto.interfaces.PBEKey() {
+            @Override
+            public String getAlgorithm() {
+                return "PBE";
+            }
+            @Override
+            public String getFormat() {
+                return null;
+            }
+            @Override
+            public byte[] getEncoded() {
+                return new byte[0];
```

Comment:
Well, instead of returning a 0-length byte array, how about throwing a RuntimeException indicating this method is not meant to be called? This would help troubleshooting if needed.

Reply:
Sounds good.